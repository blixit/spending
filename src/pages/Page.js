import React, { useEffect, useState } from 'react';
import styled from 'styled-components';

import ErrorBoundary from 'ui/structure/boundary/ErrorBoundary';
import { readCache } from 'core/storage/cache';

const StyledPage = styled.div`
  position: fixed;
  top: 50px;
  bottom: 60px;
  left: 0;
  right: 0;
  overflow-y: auto;
  width: 100%;
  background: ${({ theme }) => theme.fourth()};
  color: ${({ theme }) => theme.primary()};
  font-size: 14px;
`;

const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  padding: 20px;
`;

export default props => {
  const { router, children, ...rest } = props;
  const [canRender, setCanRender] = useState(false);
  useEffect(() => {
    const user = readCache('user');
    if (!user && router && router.location.pathname !== '/login') {
      const { history } = router;
      
      // redirect to home
      history.push('/login');
      console.log('redirect to login');
      
      return;
    }

    setCanRender(true);
  }, [router]);
  return (
    <ErrorBoundary>
      <StyledPage {...rest}>
        {canRender && <Wrapper>{children}</Wrapper>}
      </StyledPage>
    </ErrorBoundary>
  );
};
