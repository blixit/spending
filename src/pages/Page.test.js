import React from 'react';

import render from 'testing/render';

import Page from './Page';

describe('[pages] Pages', () => {
  it('should render properly', () => {
    const tree = render(<Page>Test</Page>);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
