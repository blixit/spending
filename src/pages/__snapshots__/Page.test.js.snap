// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[pages] Pages should render properly 1`] = `
<div
  className="sc-bdVaJa kjZkLq"
>
  <div
    className="sc-bwzfXH gouoTZ"
  >
    Test
  </div>
</div>
`;
